--- tracing-scenejs.js	2010-07-19 20:29:43.000000000 -0700
+++ scenejs.js	2010-07-19 21:07:11.000000000 -0700
@@ -1866,22 +1866,26 @@
     for (var i = 0; i < numUniforms; ++i) {
         var u = context.getActiveUniform(handle, i);
         if (u) {
-            var location = context.getUniformLocation(handle, u.name);
+	    u_name = u.name;
+	    if (u_name[u_name.length-1] == "\u0000") {
+		u_name = u_name.substr(0, u_name.length-1);
+	    }
+            var location = context.getUniformLocation(handle, u_name);
             if ((u.type == context.SAMPLER_2D) || (u.type == context.SAMPLER_CUBE) || (u.type == 35682)) {

-                samplers[u.name] = new SceneJS._webgl_ProgramSampler(
+                samplers[u_name] = new SceneJS._webgl_ProgramSampler(
                         context,
                         handle,
-                        u.name,
+                        u_name,
                         u.type,
                         u.size,
                         location,
                         logging);
             } else {
-                uniforms[u.name] = new SceneJS._webgl_ProgramUniform(
+                uniforms[u_name] = new SceneJS._webgl_ProgramUniform(
                         context,
                         handle,
-                        u.name,
+                        u_name,
                         u.type,
                         u.size,
                         location,